core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -62,7 +62,12 @@ public static Object getMeasureValueBasedOnDataType(String msrValue, DataType da
       case INT:
         return Double.valueOf(msrValue).longValue();
       case LONG:
-        return Long.valueOf(msrValue);
+        int index = msrValue.indexOf(".");
+        if (index != -1) {
+          return Long.valueOf(msrValue.substring(0,index));
+        } else {
+          return Long.valueOf(msrValue);
+        }
```

Comment:
@lion-x ...keep the code same for INT and LONG datatype case...i think instead of writing this code, that would be better. Sample code as below
case INT:
case LONG:
  return Double.valueOf(msrValue).longValue();